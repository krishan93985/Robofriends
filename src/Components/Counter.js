import React, { useState } from 'react';

const Counter = () => {
    const [count,setCount] = useState(0);

    console.log('Counter')
    return (
        <button onClick={() => setCount(count+1)}>Count:{count}</button>
    );
}

export default Counter;