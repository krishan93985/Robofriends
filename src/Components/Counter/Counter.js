import React, { useState } from 'react';

const Counter = () => {
    const [count,setCount] = useState(0);

    return (
        <button id="counter" data-count={count} onClick={() => setCount(count+1)} >Count:{count}</button>
    );
}

export default Counter;