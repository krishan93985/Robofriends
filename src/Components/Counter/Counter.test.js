import React from 'react';
import Counter from './Counter';
import { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';

it('is Counter component correctly rendered', () => {
    const wrapper = shallow(<Counter />);
    expect(toJson(wrapper)).toMatchSnapshot();
})

it('Count state working correctly',() => {
    const wrapper = shallow(<Counter />);
    const button = wrapper.find("[id='counter']");
    button.simulate('click');
    expect(wrapper.find("[id='counter']").props()['data-count']).toEqual(1);
})