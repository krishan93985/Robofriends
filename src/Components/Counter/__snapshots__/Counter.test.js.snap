// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`is Counter component correctly rendered 1`] = `
<button
  data-count={0}
  id="counter"
  onClick={[Function]}
>
  Count:
  0
</button>
`;
